File: algorithms/0205.isomorphic_strings/isomorphic_strings.cpp
class Solution {
public:
    void convertIsomorphic(string &s) {
        char s_map[256] = {0};
        int index = 1;
        for (int i = 0; i < s.size(); ++i) {
            const char &c = s[i];
            if (s_map[c] == 0) {
                s_map[c] = static_cast<char>(index);
                index += 1;
            }
            s[i] = s_map[c];
        }
    }
    bool isIsomorphic(string &s, string &t) {
        if (s.size() != t.size()) {
            return false;
        }
        convertIsomorphic(s);        
        convertIsomorphic(t);
        for (int i = 0; i < s.size(); ++i) {
            if (s[i] != t[i]) {
                return false;
            }
        }
        return true;
    }
};

File: algorithms/0205.isomorphic_strings/isomorphic_strings.rs
use std::collections::HashMap;

impl Solution {
    pub fn is_isomorphic(s: String, t: String) -> bool {
        let mut s_first_dict = [0; 256];
        let mut t_first_dict = [0; 256];
        let (s, t) = (s.as_bytes(), t.as_bytes());

        for i in 0..s.len() {
            if s_first_dict[s[i] as usize] != t_first_dict[t[i] as usize] {
                return false;
            }
            s_first_dict[s[i] as usize] = i + 1;
            t_first_dict[t[i] as usize] = i + 1;
        }

        true
    }
}